### DISCUSSION TITLE
Routes not set correctly in clients

### DISCUSSION THREAD

**Discussion by diegobenincasa** [ltsp/ltsp#280] [Q&A]
Hello!

I've just installed LTSP in Ubuntu 20.04, and until now everything seems fine - except that clients do not have network working correctly. Server has network working ok.

Analyzing the clients, I saw that the routing table is not right. It's almost empty, except for a single entry that maps only a subnet to a default 0.0.0.0 that is not set. When I change to an admin account and do a `sudo route add default gw x.x.x.x` (x.x.x.x being my gateway's IP) the network become alive.

Am I missing some step in LTSP setup? I followed every single step from the installation page.

Notice: I have an external DHCP server.

Thanks in advance.

**Comment by alkisg** [comment]
Hello, for Debian-based netbooted PCs, the package "initramfs-tools" sets up networking, so this issue isn't related to LTSP.

I believe that you're just missing an `option routers xxx` entry in your dhcpd.conf.

**Comment by diegobenincasa** [comment]
I already have this option in my dhcpd.conf, since this DHCP server was already on before LTSP and works fine for every non-LTSP machine on the network.

**Comment by alkisg** [comment]
In non-netbooted clients, networking is handled by network-manager or netplan or ifupdown etc. So initramfs-tools might not like something that network-manager doesn't have a problem with. For example, network-manager might automatically use router=dhcp-server, when no router is sent.

Is this a chrootless setup? If so, which method do you use to configure the LTSP server's networking?

**Comment by diegobenincasa** [comment]
Yes, it's a chrootless setup.

The server was configured using Network Manager (GUI) and static IP. Should I change the network setup procedure from NetworkManager to editing `/etc/network/interfaces` (by installing `ifupdown` package)?

**Comment by alkisg** [comment]
No, network-manager is fine (and ifupdown is fine too). Boot up a client, open a terminal, see the interface name, and then run the following, and finally upload the output here:

```shell
cat /run/net-enp2s0.conf
/usr/lib/klibc/bin/ipconfig -n enp2s0
```

You may also come to the IRC channel if you want.

**Comment by alkisg** [comment]
The problem was that when a DHCP server sends both a gateway and some routes, `ipconfig` only remembers the routes.

So /run/net-enp2s0.conf gets `IPV4ROUTE0GATEWAY='1.2.3.4'` instead of `IPV4GATEWAY='1.2.3.4'`.

If someone needs a proper fix, he should file a bug against klibc. An easy workaround is to use a POST_INIT_SET_GATEWAY="" parameter in ltsp.conf.
